### PR TITLE
Legacysettings telemetry

### PR DESCRIPTION
**Pull request by singhsarab**
Adding the telemetry points for
- Testsettings usage
- Elements inside LegacySettings

### CODE REVIEW

**Review by kaadhina — APPROVED** [review]

File: src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs [hunk 1]
```diff
@@ -285,6 +285,47 @@ public static bool AreRunSettingsCollectorsInCompatibleWithTestSettings(string r
...
+
+                    while (childNodes.MoveNext())
+                    {
+                        legacySettings.Add(childNodes.Current.Name);
+                    }
+                }
+            }
+            catch(Exception ex)
+            {
+                EqtTrace.Error("Error while trying to read legacy settings. Message:", ex.Message);
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs]: Add `{0}` and print `ex.ToString()`.

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 2]
```diff
@@ -287,6 +290,16 @@ public void RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLaunc
             }
         }
 
+        private void LogTelemetryForLegacySettings(IRequestData requestData, string runsettings)
+        {
+            requestData.MetricsCollection.Add(TelemetryDataConstants.TestSettingsUsed, InferRunSettingsHelper.IsTestSettingsEnabled(runsettings));
+            
+            if (InferRunSettingsHelper.TryGetLegacySettings(runsettings, out List<string> legacySettings))
```
Comment (smadala) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Nit: Rename `legacySettings` to `legacySettingsElements`.

File: test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs [hunk 3]
```diff
@@ -950,6 +950,99 @@ public void RunTestsShouldCollectCommands()
...
+
+            // Verify
+            Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.LegacySettings, out var legacySettingsString));
+            StringAssert.Equals("Deployment, Scripts", legacySettingsString);
+
+            Assert.IsTrue(actualRequestData.MetricsCollection.Metrics.TryGetValue(TelemetryDataConstants.TestSettingsUsed, out var testSettingsUsed));
+            Assert.IsFalse((bool)testSettingsUsed);
+
+            // Opt out the Telemetry
+            Environment.SetEnvironmentVariable("VSTEST_TELEMETRY_OPTEDIN", "0");
```
Comment (smadala) [inline test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs]: Do this in clean up.

File: src/Microsoft.TestPlatform.Common/Telemetry/TelemetryDataConstants.cs [hunk 4]
```diff
@@ -17,6 +17,10 @@ public static class TelemetryDataConstants
         // Total time taken to complete one test run request
         public static string TimeTakenInSecForRun = "VS.TestRun.TimeTakenInSec";
 
+        public static string TestSettingsUsed = "VS.TestRun.TestSettingsUsed";
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Common/Telemetry/TelemetryDataConstants.cs]: Nit: Rename this to `IsTestSettingsUsed`.

**Review by smadala — APPROVED** [review]

**Comment by smadala** [comment]
Add link to user story.

**Pull request merged**